Assets/Scripts/Modules/Turn/ITurnModule.cs

```diff
@@ -1,8 +1,17 @@
 ﻿/*
- * Authors: Xijie Guo, Isaiah Mann
+ * Authors: Xijie Guo
  * Description: An interface to handle the turn logic
  */
 
-public interface ITurnModule {
+public interface ITurnModule { 
 
+	void SwitchToMonTurn();
+
+	bool CheckMonsterAttack(IUnit monster, IUnit player);
```

Comment:
Have you accounted for handling melee vs magic attacks? Also, can we slightly rename this to be more expressive, I assume the intent is something like:

"CheckIfMonsterCanAttack(IUnit monster, IUnit player);"